File: webshared/src/main/scala/com/taskadapter/webui/Page.scala
package com.taskadapter.webui

import com.taskadapter.web.data.Messages

object Page {
  private val BUNDLE_NAME = "com.taskadapter.webui.data.messages"
  val MESSAGES = new Messages(BUNDLE_NAME)

  def message(key: String): String = MESSAGES.get(key)

  //  @annotation.varargs is a hint to generate a Java version of the method which takes an array instead of a Seq.
  @annotation.varargs def message(key: String, argument: String*): String = MESSAGES.format(key, argument)

  def message(key: String, argument: String): String = MESSAGES.format(key, argument)
  def message(key: String, argument1: String, argument2: String): String = MESSAGES.format(key, argument1, argument2)
}

File: webshared/src/main/java/com/taskadapter/webui/Page.java
package com.taskadapter.webui;

import com.taskadapter.web.data.Messages;

public class Page {
    private static final String BUNDLE_NAME = "com.taskadapter.webui.data.messages";
    public static final Messages MESSAGES = new Messages(BUNDLE_NAME);

    public static String message(String key) {
        return MESSAGES.get(key);
    }

    public static String message(String key, String... arguments) {
        return MESSAGES.format(key, (Object[]) arguments);
    }

    public static String message(String key, String argument) {
        return MESSAGES.format(key, argument);
    }

    public static String message(String key, String argument1, String argument2) {
        return MESSAGES.format(key, argument1, argument2);
    }
}
